transport/tchannel/outbound.go

```diff
@@ -133,10 +133,17 @@ func callWithPeer(ctx context.Context, req *transport.Request, peer *tchannel.Pe
 		return nil, err
 	}
 
+	var headerItems map[string]string
+	if rawHeader {
+		headerItems = req.Headers.ForwardingItems()
+	} else {
+		headerItems = req.Headers.Items()
+	}
```

Comment:
Can we do this in a way that doesn't require copying this logic? I don't want to have to hunt down every place where `rawHeader` is inspected if a change is needed to this behavior. Ideally this happens in one place only. writeHeaders or writeRequestHeaders or one of those.